### PR TITLE
Architecture: Cleanup Integration Tests

### PR DESCRIPTION
**Pull request by talekhinezh**
* Introduce notion of "Physical Nodes" as opposed to BFTNodes in Integration test
* Cleanup `DeterministicTest.java`

### CODE REVIEW

**Review by jakrawcz-rdx — COMMENTED** [review]
A test cleanup is always welcome 👍

File: core/src/test-core/java/com/radixdlt/consensus/MockedEpochsConsensusRecoveryModule.java [hunk 1]
```diff
@@ -69,58 +69,80 @@
...
+            epoch ->
+                BFTValidatorSet.from(
+                    epochNodeWeightMapping
+                        .nodesAndWeightFor(epoch)
+                        .map(
+                            niw ->
+                                BFTValidator.from(
+                                    BFTNode.create(
+                                        PrivateKeys.ofNumeric(niw.index() + 1).getPublicKey()),
+                                    niw.weight()))));
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/test-core/java/com/radixdlt/consensus/MockedEpochsConsensusRecoveryModule.java:123]: Maybe "cleanup" is the wrong word (more like "refactor to enable better testing") but might as well :) Will clean this part up a bit.

File: core/src/test-core/java/com/radixdlt/environment/deterministic/network/ControlledSender.java [hunk 2]
```diff
@@ -70,15 +70,22 @@
...
 public final class ControlledSender implements Environment {
   private final DeterministicNetwork network;
   private final BFTNode self;
   private final int senderIndex;
   private final ChannelId localChannel;
+  private final Function<BFTNode, Integer> addressBook;
 
-  ControlledSender(DeterministicNetwork network, BFTNode self, int senderIndex) {
+  public ControlledSender(
+      Function<BFTNode, Integer> addressBook,
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/test-core/java/com/radixdlt/environment/deterministic/network/ControlledSender.java:84]: Yup, exactly. I agree with updating `Integer` to something more meaningful but would rather push this update to a few later PRs as we figure out the right abstraction.

File: core/src/test-core/java/com/radixdlt/harness/deterministic/DeterministicNodes.java [hunk 3]
```diff
@@ -98,21 +100,27 @@
 public final class DeterministicNodes implements AutoCloseable {
   private static final Logger log = LogManager.getLogger();
 
+  // Nodes
   private final List<Injector> nodeInstances;
-  private final DeterministicNetwork network;
-  private final ImmutableBiMap<BFTNode, Integer> nodeLookup;
-
+  private final Map<BFTNode, Integer> addressBook;
+  private final Map<Integer, BFTNode> nodeIdentifiers;
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/test-core/java/com/radixdlt/harness/deterministic/DeterministicNodes.java:106]: These two are different abstractions in that the `addressBook` is used for resolving logical ids to physical ids of nodes. `nodeIdentifiers` is used for a physical node's configuration. The fact that a `BiMap` could represent both abstractions is pattern matching taking advantage of the fact that the current abstraction of tests don't distinguish between the two.

Admittedly, this is a little bit abstract in this PR but becomes more apparent in a later PR (see branch `feature/validator-component`).

[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply core/src/test-core/java/com/radixdlt/harness/deterministic/DeterministicNodes.java:106]: ahh, thanks for the explanation! And I really appreciate splitting big changes into multiple PRs 👍

File: core/src/integration/java/com/radixdlt/integration/steady_state/simulation/consensus/UniformLatencyTest.java [hunk 4]
```diff
@@ -93,7 +95,13 @@ public void given_4_correct_bfts__then_should_pass_sanity_tests_over_1_minute()
...
         SimulationTest.builder()
             .networkModules(NetworkOrdering.inOrder(), NetworkLatencies.fixed())
-            .numNodes(4)
+            .numPhysicalNodes(4)
+            .functionalNodeModule(
+                new FunctionalRadixNodeModule(
+                    false,
+                    SafetyRecoveryConfig.mocked(),
+                    ConsensusConfig.of(),
+                    LedgerConfig.mocked(4)))
```
Comment (dhedey) [inline core/src/integration/java/com/radixdlt/integration/steady_state/simulation/consensus/UniformLatencyTest.java:104]: I'm not sure what this parameter means, perhaps `mocked` could do with a slightly better name?

**Review by dhedey — APPROVED** [review]
I'm approving - however a bit worried about how it's significantly harder to add full nodes to a test now - thoughts below

Perhaps this abstraction can be adjusted in a later PR?

-------

A lot of these tests previously both created validators and full nodes in their `numNodes` property - and we seem to have lost the distinction in these tests now... It appears for most tests we are now assuming all nodes are validators? This reduces coverage of some tests, although I see you've explicitly created full nodes in tests which test mempool sync from a full node to a validator.

I do like the idea of configuring the node/network configuration explicitly in a call to the builder - as understanding how to set V validators and F full nodes requiring specifying parameters as `physicalNodes: V + F`, and a few random ledger config and recovery parameters as `V` is going to be so easy to mess up, or more to the point give up, and just forget about full nodes.

I'd like it to be obvious to configure this network arrangement, which ideally is respected automatically by the functional node module; unless there's explicit overriding?

Perhaps `numPhysicalNodes` actually needs to be a slightly more complicated abstraction, eg `withNodeSetup`, allowing us to set physical node arrangement and the initial validator stake there easily? (with some easily helper abstractions) - and then the other configs read from this in the builder as a default, unless some override is specified?

**Comment by talekhinezh** [comment]
> It appears for most tests we are now assuming all nodes are validators?

This isn't true or atleast isn't the intention. The distinction is still there but is just managed at the state computer layer rather than by the test "physical environment". The `withNodeSetup` is exactly the approach taken in follow up PRs.

**Pull request merged**